cmd/agent/entrypoint.go

```diff
@@ -67,52 +94,28 @@ func NewEntrypoint(logger *util.Logger, cfg *config.Config) (*Entrypoint, error)
 	if err := ep.ApplyConfig(*cfg); err != nil {
 		return nil, err
 	}
-	return &ep, nil
+	return ep, nil
 }
 
 // ApplyConfig applies changes to the subsystems of the Agent.
 func (srv *Entrypoint) ApplyConfig(cfg config.Config) error {
 	srv.mut.Lock()
 	defer srv.mut.Unlock()
 
-	// The server config uses some unexported fields which can't be compared by
-	// default. Since only exported fields are used by YAML, we'll only compare
-	// those here.
-	var ignoreUnexported = cmpopts.IgnoreUnexported(logging.Format{}, logging.Level{})
-
-	if cmp.Equal(srv.cfg, cfg, ignoreUnexported) {
-		return nil
+	if cfg.Server.Log == nil {
```

Comment:
Do these two lines need to be reversed? IE set the src.cfg.Server.Log?

Reply:
I pushed this logic to the caller to make it a little less awkward.